config/initializers/rack_attack.rb

```diff
@@ -62,28 +62,23 @@ class Rack::Attack
   login_limit = ArchiveConfig.RATE_LIMIT_LOGIN_ATTEMPTS
   login_period = ArchiveConfig.RATE_LIMIT_LOGIN_PERIOD
 
-  # Throttle POST requests to /login by IP address
+  # Throttle POST requests to /users/login by IP address
   #
   # Key: "rack::attack:#{Time.now.to_i/:period}:logins/ip:#{req.ip}"
-  throttle('logins/ip', limit: login_limit, period: login_period) do |req|
-    if req.path == '/login' && req.post?
-      req.ip
-    end
+  throttle("logins/ip", limit: login_limit, period: login_period) do |req|
+    req.ip if req.path == "/users/login" && req.post?
   end
 
-  # Throttle POST requests to /login by email param
+  # Throttle POST requests to /users/login by login param (user name or email)
   #
-  # Key: "rack::attack:#{Time.now.to_i/:period}:logins/email:#{req.email}"
+  # Key: "rack::attack:#{Time.now.to_i/:period}:logins/email:#{login}"
```

Comment:
Question! Why is it `#{login}` in the key instead of `#{req.login}`?

Reply:
`req.login` is not actually field. Neither was `req.email`, which was copied from Rack::Attack's examples. The value should be `req.params.dig("user", "login").presence` but I didn't want to repeat that in the comment.